src/kakarot/execution_context.cairo

```diff
@@ -173,18 +184,33 @@ namespace ExecutionContext {
     // @param len The size of the data to read.
     // @return The pointer to the updated execution context.
     // @return The data read from the bytecode.
-    func read_code(self: model.ExecutionContext*, len: felt) -> (
-        self: model.ExecutionContext*, output: felt*
+    func read_code{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(self: model.ExecutionContext*, len: felt) -> (
+        self: model.ExecutionContext*, stack_element: Uint256
```

Comment:
why do we change the signature?

Reply:
This is only being used by Push opcode, which basically needs an uint256